test/tests/functional/pbs_qsub_direct_write.py

```diff
@@ -106,3 +230,48 @@ def test_qalter_direct_write(self):
                                    id=jid)
         except PbsAlterError as e:
             print str(e)
+
+    def test_qalter_direct_write_error(self):
+        """
+        submit a job and after it starts running alter
+        the job with -koed and check whether expected
+        error message appears
+        """
+        j = Job(TEST_USER)
+        jid = self.server.submit(j)
+        attribs = {ATTR_k: 'oed'}
+        self.server.expect(JOB, {'job_state': 'R'})
+        try:
+            self.server.alterjob(jid, attribs)
+        except PbsAlterError as e:
+            self.assertTrue(
+                'qalter: Cannot modify attribute while job running  Keep_Files'
```

Comment:
are you sure that when run is API mode it will spit out an error prepended with "qalter: " in  it?
Just remove it if it is not important. Are you also sure there will be 2 spaces between the error message and attribute name ("Keep_Files")?

Reply:
Yes there are 2 spaces